tailcfg/tailcfg.go

```diff
@@ -223,6 +224,12 @@ type Hostinfo struct {
 	RoutableIPs   []wgcfg.CIDR `json:",omitempty"` // set of IP ranges this client can route
 	Services      []Service    `json:",omitempty"` // services advertised by this machine
 
+	// Network conditions:
+	MappingVariesByDestIP opt.Bool
+	WorkingIPv6           opt.Bool
+	WorkingUDP            opt.Bool
```

Comment:
Some forward-looking thoughts (even though we get a chance to reinvent this data structure I guess when we do control protocol v2):

- some networks block UDP entirely.
- some allow port 53
- some allow port 443 (potentially useful when people put tailscale on a machine with a real IP: we could serve out of 443 if they're not using it and people on nasty networks could skip DERP)

Reply:
Ack